src/vos/vos_dtx.c

```diff
@@ -1519,10 +1553,23 @@ vos_dtx_abort(daos_handle_t coh, daos_epoch_t epoch, struct dtx_id *dtis,
 	/* Abort multiple DTXs via single PMDK transaction. */
 	rc = vos_tx_begin(vos_cont2umm(cont));
 	if (rc == 0) {
-		for (i = 0; i < count; i++)
-			vos_dtx_abort_one(cont, epoch, &dtis[i]);
+		int	aborted = 0;
 
-		rc = vos_tx_end(vos_cont2umm(cont), 0);
+		for (i = 0; i < count; i++) {
+			rc = vos_dtx_abort_one(cont, epoch, &dtis[i]);
+			if (rc == 0)
+				aborted++;
+		}
+
+		/* Some vos_dtx_abort_one may hit failure, for example, not
+		 * found related DTX entry in the active DTX table, that is
+		 * not important, go ahead. Because each DTX is independent
+		 * from the others. For the DTXs that have been aborted, we
+		 * cannot re-insert them back into the active DTX table (in
+		 * DRAM) even if we abort this PMDK transaction, then let's
+		 * commit the PMDK transaction anyway.
+		 */
```

Comment:
We can't force transaction commit in every failure case, I think the correct way is to:
- Make DRAM changes on commit callback, or;
- Rollback DRAM changes on transaction abort;

But I'm fine to land this patch as-is.

Reply:
We maintain the active DTX table in DRAM, for batched commit case, some DTX entries may have been removed from the in-DRAM active DTX table before current DTX entry failed to be committed. If we want to abort, be as the rollback, we have to explicitly re-insert former committed DTX entries back into the active DTX table, but we cannot guarantee the correctness of such rollback. So the most direct solution is that: commit current PMDK transaction, then former committed ones will be finial committed, the others DTX entries will be committed next time.